### ISSUE TITLE
git version 1.8.3.1 does not support git -C which the latest version of gitleaks uses. date(2023-11-30).  

### ISSUE THREAD

**Issue by jarbassaidai** [gitleaks/gitleaks#1305]
**Describe the bug**
when issuing the command gitleaks detect .  the following errors occur:
ERR [git] unknown option -C
ERR [git] usage ....

**To Reproduce**
latest version of gitleaks as of 2023-11-30
version 
git version 1.8.3.1

cd to any git repository 
gitleaks detect . -v    
ERR [git] unknown option -C 

see above 
**Expected behavior**
gitleaks scan should take place with normal  out without git errors

**Screenshots**

**Basic Info (please complete the following information):**
 - OS: ubuntu 18.04 
 - Gitleaks Version: latest as of 2023-11-30

**Additional context**
sources/git.go has the error I was able to work around the problem by changing -C to -c on lines around 31, 50, 89  then rebuilding the code. 

cc @zricethezav

**Comment by rgmz** [comment]
[Git v1.8.3.1](https://github.com/git/git/releases/tag/v1.8.3.1) was released in 2013. It would probably be easier to upgrade to a newer version of Git than rewrite how Gitleaks uses Git.

**Comment by jarbassaidai** [comment]
agreed, unfortunately for me this a corp. and they don't update very often.  :)
